pilot/pkg/networking/core/v1alpha3/envoyfilter/cluster_patch.go

```diff
@@ -75,13 +75,61 @@ func mergeTransportSocketCluster(c *cluster.Cluster, cp *model.EnvoyFilterConfig
 		return false, fmt.Errorf("cast of cp.Value failed: %v", okCpCast)
 	}
 
+	merged = false
+
+	// First merge transport socket matches with the same name
+	if (len(c.GetTransportSocketMatches())) > 0 && (len(cpValueCast.GetTransportSocketMatches()) > 0) {
```

Comment:
Can we merge this loop with L133

Reply:
I can try, let me see.

Reply:
It should be possible, but I would need to remove check in L127.
Doing it separately should avoid clutter and workarounds for making this integrated.

I added some comments to the whole process.